### PR TITLE
Add semantic icon metadata pipeline with Weather Icons pilot

### PR DESCRIPTION
**Pull request by amorimjj**
## Summary

This PR introduces the first semantic icon-context pipeline for Rocketicons.

It adds bilingual metadata that helps LLMs and search users find icons by intent and UI context, rather than relying only on icon names. For example, searches such as `insurance`, `extraterrestre`, `baixar`, or `ferrovia` can resolve relevant icons.

## Pilot scope

Metadata was generated for **Weather Icons only**:

- 219 icons
- 171 semantic families
- English and Brazilian Portuguese
- Complete coverage with no missing, stale, or orphaned icons

Weather Icons was intentionally selected because it is small enough to review but diverse enough to validate weather, alerts, measurements, navigation, astronomy, transportation, actions, and other UI contexts.

**We should not generate metadata for every collection yet.** First, we need to confirm that the schema, file locations, taxonomy, Algolia attributes, search behavior, review process, and deployment integration are correct. Changes become considerably more expensive after metadata is generated for all 46,000+ icons.

## What changed

- Added versioned semantic context envelopes under `/ai/v1/`.
- Added explicit `none`, `single`, and `chunked` storage modes.
- Enforced a 64 KiB maximum for source and public JSON files.
- Added coverage reporting and SHA-256 checksums.
- Added English/PT-BR descriptions, aliases, search terms, categories, UI contexts, roles, visual properties, and negative guidance.
- Added source and prompt hashes so unchanged icons are skipped and changed icons become stale.
- Added resumable local generation and confirmation gates for large or forced regeneration.
- Added a repository-local skill for manually generating and reviewing metadata.
- Updated `llms.txt`, deployment documentation, and the AI roadmap.

## Algolia

Algolia records now consume the reviewed metadata and include:

- `descriptionEn`
- `descriptionPtBr`
- `searchTermsEn`
- `searchTermsPtBr`
- `categories`
- `uiContexts`
- `roles`
- `variant`

`negativeTerms` are intentionally not indexed as positive search terms.

The local generation skill only runs:

```bash
npm run indexer -- --dry-run
```

This validates the complete future record set without updating the live index.

Production Algolia synchronization remains restricted to:

- A push/merge to `main` after the complete build and Cloudflare deployment.
- An explicitly requested manual workflow run on `main`.

Pull-request builds do not receive the Algolia indexing key and cannot update the production index.

## Algolia review requested

@programad, since you worked actively on the Algolia integration, please pay special attention to:

- Searchable attribute ordering and priority.
- The new facets and their intended use.
- English and Portuguese fields in the same icon record.
- Record-size limits.
- Index-version calculation and synchronization skipping.
- `replaceAllObjects` behavior.
- Whether synchronization is sufficiently restricted to full `main` deployments.
- Whether aliases should remain combined with search terms or become separate attributes.

## Before expanding to other collections

Please review whether:

- The metadata lives in the right package and directory.
- The public envelope structure is appropriate for long-term compatibility.
- The taxonomy is sufficiently stable.
- Family-level generation with per-icon materialization is the right balance.
- The 64 KiB limit and topic-based chunk descriptors are useful to LLM clients.
- Partial coverage behavior is correct.
- Algolia ranking produces relevant results without weakening existing name-based search.
- The manual review and force-confirmation gates are sufficient.

Only after these points are settled should we start generating metadata for additional collections.

## Verification

- 38 tests passing.
- TypeScript compilation passing.
- ESLint passing.
- Full static generation completed for 46,566 icons across 32 collections.
- Algolia dry run validated 46,657 records.
- Weather Icons coverage verified at 219/219.
- Public JSON Schemas validated against generated artifacts.
- Public chunk sizes and SHA-256 checksums verified.
- Incremental preparation verified with zero unchanged icons regenerated.
- Forced regeneration verified to require explicit confirmation.

## Deployment note

This PR does not update the live Algolia index. The enriched records will be synchronized only after the change reaches `main` through the normal full deployment workflow.

### CODE REVIEW

File: packages/ignition/src/icon-context/core.ts [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+const makeEnvelope = (collectionId: string, chunk: number, icons: PublicIconContext[]) =>
+  ({
+    schemaVersion: ICON_CONTEXT_SCHEMA_VERSION,
+    kind: "rocketicons.icon-context-data",
+    collectionId,
+    chunk,
+    icons
+  }) satisfies ContextDataEnvelope;
+
+const splitByBytes = (collectionId: string, icons: PublicIconContext[]) => {
```
Comment (programad) [inline packages/ignition/src/icon-context/core.ts:262]: `splitByBytes` only splits when there is already an icon in the current group; it never validates the size of a one-icon public envelope. A valid source family at the 64 KiB limit can materialize into a larger public envelope because the latter adds `variant` and `visual`. I reproduced a valid 65,536-byte source producing a 65,634-byte public chunk, which the build would publish despite the advertised hard limit. Please validate every emitted public chunk (and add this one-entry boundary case to the tests).

File: packages/ignition/icon-context/wi/1.json [hunk 2]
```diff
@@ -0,0 +1 @@
...","forecast","protection","coverage","insurance"],"pt-BR":["guarda-chuva","chuva","previsão do tempo","proteção","cobertura","seguro"]},"negativeTerms":{"en":[],"pt-BR":[]},"primaryCategory":"weather","categories":["weather","objects"],"uiContexts":["forecast","travel","insurance"],"roles":["status"],"icons":[{"id":"wi-umbrella","sourceHash":"32c762e141399ce56b98abfc62e3dfcbcc5ddcc1801a3deb6281741ace9bf017"}]},{"familyId":"volcano","description":{"en":"A volcano symbol for emergency alerts, hazard maps, safety notices, and severe-weather tracking.","pt-BR":"Um símbolo de vulcão para alertas de emergência, mapas de risco, avisos de segurança e clima severo."},"aliases":{"en":[],"pt-BR":[]},"searchTerms":{"en":["volcano","natural disaster","emergency","hazard","warning","safety alert"],"pt-BR":["vulcão","desastre natural","emergência","risco","alerta","alerta de segurança"]},"negativeTerms":{"en":[],"pt-BR":[]},"primaryCategory":"disasters","categories":["disasters","alerts","weather"],"uiContexts":["alerts","notifications","maps","forecast","emergency"],"roles":["status"],"icons":[{"id":"wi-volcano","sourceHash":"30c60241b27df6a1df3087bcdcde86d2e92ed0b72a444d75397d469dd4101f33"}]}]}
```
Comment (programad) [inline packages/ignition/icon-context/wi/1.json]: The `train` family has English aliases `precipitation` / `rainfall` and the equivalent Portuguese aliases, while its terms correctly describe rail transport. This will make the train icon rank for rain queries and undermines the pilot’s semantic-search validation. Please replace these with train-specific aliases or remove them.

**Review by programad — CHANGES_REQUESTED** [review]
Requesting changes before merging this pilot: the public context builder can emit a chunk larger than its advertised 64 KiB hard limit, and the `wi-train` metadata incorrectly maps rain-related aliases to a train icon. The production Algolia synchronization guard otherwise looks correctly restricted to `main`.

File: packages/ignition/src/icon-context/core.ts (hunk 1 above)
Comment (amorimjj) [reply packages/ignition/src/icon-context/core.ts:262]: Fixed in f82100ae. We now validate the final public envelope after chunking, including the single-icon case, and fail if any emitted file exceeds the 64 KiB limit. I also added a regression test reproducing the exact boundary condition you reported.

File: packages/ignition/icon-context/wi/1.json (hunk 2 above)
Comment (amorimjj) [reply packages/ignition/icon-context/wi/1.json]: Fixed in 81dec54c. `wi-train` now uses `locomotive` / `locomotiva aliases and contains no rain or precipitation metadata.
We also addressed the broader cause rather than special-casing this icon: metadata generation must now use complete ID tokens, every family must be audited in both languages, aliases shared across unrelated categories are rejected, and incremental runs validate the complete merged collection. We re-audited all 171 Weather Icons families under the updated prompt contract.

**Review by programad — APPROVED** [review]
